tests/src/test/java/alluxio/master/journal/ufs/UfsJournalLogWriterTest.java

```diff
@@ -178,6 +197,229 @@ public void writeJournalEntryRotate() throws Exception {
+    writer.write(newEntry(nextSN));
+    writer.close();
+  }
+
+  @Test
+  public void missingIncompleteJournalFile() throws Exception {
+    long startSN = 0x10;
+    UfsJournalLogWriter writer = new UfsJournalLogWriter(mJournal, startSN);
+    long nextSN = writeJournalEntries(writer, startSN, 5);
+    // Create a mock DataOutputStream.
+    DataOutputStream badOut = createMockDataOutputStream(writer);
+    // Specify the behavior of badOut so that it throws an IOException containing
+    // "injected I/O error" when its `write` method is invoked with any arguments
+    // matching (byte[], int, int).
+    Mockito.doThrow(new IOException(INJECTED_IO_ERROR_MESSAGE)).when(badOut)
+        .write(Mockito.any(byte[].class), Mockito.anyInt(), Mockito.anyInt());
+    tryWriteAndExpectToFail(writer, nextSN);
+    UfsJournalSnapshot snapshot = UfsJournalSnapshot.getSnapshot(mJournal);
+    UfsJournalFile journalFile = snapshot.getCurrentLog(mJournal);
+    File file = new File(journalFile.getLocation().toString());
+    file.delete();
+    mThrown.expect(RuntimeException.class);
+    mThrown.expectMessage(
+        ExceptionMessage.JOURNAL_ENTRY_MISSING.getMessageWithUrl(
+            RuntimeConstants.ALLUXIO_DEBUG_DOCS_URL, 0, 0x10));
+    writer.write(newEntry(nextSN));
+    writer.close();
+  }
+
+  /**
+   * Writes {@code numberOfEntries} journal entries starting from {@code startSN}.
+   *
+   * @param writer {@link UfsJournalLogWriter} instance to use
+   * @param startSN start journal sequence number
+   * @param numberOfEntries number of journal entries
+   * @return next journal entry sequence number after writing these entries
+   */
+  private long writeJournalEntries(UfsJournalLogWriter writer, long startSN,
+      int numberOfEntries) throws IOException {
+    long nextSN = startSN;
+    for (int i = 0; i < numberOfEntries; i++) {
+      writer.write(newEntry(nextSN));
+      nextSN++;
+    }
+    Assert.assertNotNull(writer.getJournalOutputStream());
+    return nextSN;
+  }
+
+  /**
+   * Creates a mock {@link DataOutputStream} for {@link UfsJournalLogWriter#mJournalOutputStream}.
+   *
+   * @param writer the {@link UfsJournalLogWriter} instance for which the mock is created
+   * @return the created mock {@link DataOutputStream} instance
+   */
+  private DataOutputStream createMockDataOutputStream(UfsJournalLogWriter writer) {
+    DataOutputStream badOut = Mockito.mock(DataOutputStream.class);
+    Object journalOutputStream = writer.getJournalOutputStream();
+    // Use Whitebox to set the private member of journalOutputStream to badOut.
+    Whitebox.setInternalState(journalOutputStream, "mOutputStream", badOut);
+    Assert.assertEquals(badOut, writer.getUnderlyingDataOutputStream());
+    return badOut;
+  }
+
+  /**
+   * Tries to write a journal entry and expects the write to fail due to
+   * {@link IOException} thrown by {@link DataOutputStream#write(byte[], int, int)}.
```

Comment:
nothing in here is related to `DataOutputStream`

Reply:
Done.